src/runtime/internal/dom.ts

```diff
@@ -644,6 +644,26 @@ export function query_selector_all(selector: string, parent: HTMLElement = docum
 	return Array.from(parent.querySelectorAll(selector)) as ChildNodeArray;
 }
 
+export function get_hydratable_raw_elements(parent: HTMLElement = document.body) {
+	const elements = Array.from(parent.childNodes);
+	let cursor = 0;
+	while (cursor < elements.length) {
+		const start_index = find_comment(elements, 'HTML_TAG_START', cursor);
+		const end_index = find_comment(elements, 'HTML_TAG_END', start_index);
+		if (start_index === end_index) {
+			elements.splice(cursor, elements.length - cursor);
+			break;
+		}
```

Comment:
what happends if you have multiple `{@html 'xxx'}` in the head?

how woudl you ensure you are not claiming the html tag of other components?